src/test/java/StringManipulationTest.java

```diff
@@ -28,58 +28,89 @@ public void testCount1() {
 
     @Test
     public void testCount2() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("Thisismystring");
+        int length = manipulatedstring.count();
+        assertEquals(1, length);
     }
 
     @Test
     public void testCount3() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("23 and me");
+        int length = manipulatedstring.count();
+        assertEquals(3, length);
     }
 
     @Test
-    public void testCount4() {
-        fail("Not yet implemented");
+    public void testCount4() { 
+    	manipulatedstring.setString("This is  my string");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);   
+    }
+    
+    @Test
+    public void testCount5() { 
+    	manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);   
     }
 
     @Test
     public void testRemoveNthCharacter1() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
+        String temp =  manipulatedstring.removeNthCharacter(3, false);
+        assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", temp);
     }
 
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        String temp =  manipulatedstring.removeNthCharacter(3, true);
+        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", temp);
     }
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+    	String temp = manipulatedstring.removeNthCharacter(5, false);    
+    	
+    	assertEquals("I'd 3tt3 puts0med161s inthis5tr16, rght?", temp);
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertEquals("I'd  3tt3  put s0me d161 s in this 5tr1 6, r ght?", manipulatedstring.removeNthCharacter(5, true));    
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+    	Throwable exception = assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.removeNthCharacter(100, false));
```

Comment:
I believe you can use the assertThrows() method without having to set it equal to a Throwable object. The assertThrow() method checks to see whether the method you provided in the lambda function returns the same Throwable as the Throwable you claim it will throw. Your way of checking definitely works.